index-server/load-test-app/Main.hs

```diff
@@ -0,0 +1,125 @@
+module Main where
+
+import Control.Concurrent
+import Control.Monad
+import Control.Monad.Trans.Except
+import Data.ByteString.Builder
+import Data.Time.Clock.POSIX
+import Data.Word
+import Ergvein.Index.Protocol.Serialization
+import Ergvein.Index.Protocol.Utils
+import Ergvein.Index.Server.TCPService.Socket (sendLazy)
+import Foreign.C.Types
+import Network.Socket hiding (recv)
+import Network.Socket.ByteString  (recv, sendAll)
+import Options.Applicative
+import System.Random
+import Text.Read
+
+import Ergvein.Index.Protocol.Deserialization
+import Ergvein.Index.Protocol.Types
+import Ergvein.Index.Server.TCPService.Server
+import Ergvein.Index.Server.TCPService.Supervisor
+import Ergvein.Text
+
+import qualified Control.Exception           as E
+import qualified Data.ByteString             as BS
+import qualified Data.Text.IO                as T
+import qualified Data.Vector.Unboxed         as UV
+
+filterBatchSize, filterStartHeight, filterEndHeight :: Word64
+filterBatchSize   = 300
+filterStartHeight = 400000
+filterEndHeight   = 550000
+
+data Options = Options {
+  -- | Which command to execute
+    optsCommand           :: Command
+  -- |Def: False. Start only BC-scanning threads
+}
+
+wordReader :: ReadM Word64
+wordReader = eitherReader $ \arg -> case readMaybe arg of
+  Nothing -> Left ("Cannot parse word: " ++ arg)
+  Just w  -> Right w
+
+data Command = CommandLoad Int String String
+
+options :: Parser Command
+options = subparser (command "listen" $ info (loadCmd <**> helper) $ progDesc "Start server")
+  where
+    loadCmd = CommandLoad
+            <$> (read <$> strArgument ( metavar "CONFIG_N_CLIENTS" ))
+            <*> strArgument ( metavar "CONFIG_ADDR" )
+            <*> strArgument ( metavar "CONFIG_PORT" )
+
+startServer :: Command -> IO ()
+startServer cmd = case cmd of
+  CommandLoad clientsNumber addr port -> do
+    T.putStrLn "Server starting"
+    withWorkersUnion $ \wrk -> do
+      replicateM_ clientsNumber $ spawnWorker wrk $ forever $ runTCPClient addr port $ onException . receiveFilters 
+      forever $ threadDelay maxBound
+  where
+    onException :: IO () -> IO ()
+    onException = (`E.catch` (print . show :: E.SomeException -> IO ()))
```

Comment:
It's better to pick another name in order to avoid name clash with `onException` from Control.Exception. It could be confusing for reader.